autoshop/core/views.py

```diff
@@ -74,10 +76,10 @@ def log_in(request):
                 password=form.data['password'],
             )
             if not user:
-                return render(request, 'failure.html')
+                raise PermissionDenied
```

Comment:
Again we will redirect the user to login page with error message. We can do that with django messages.